### PR TITLE
Misc: Generate and push repros to a GitHub repo every night 

### PR DESCRIPTION
**Pull request by gaetanmaisse**
## What I did

Added a script to generate and push repros every night. It is using
 - `sb repro` from the Storybook CLI
 - [zx](https://github.com/google/zx), I discovered it a few weeks (months?) ago and had the opportunity to use it for non-trivial script and I find it really pleasant to use, it simplifies a lot the syntax when writing JS scripts. It could be a great lib to replace all the `cross-spawn`, `spawn` etc usages and our custom `exec` functions we have in the CLI and in the tooling scripts. To discuss with other maintainers 😄 

## TODO

 - Update the READMEs
 - Update the GH user used in the GH Action + generate a PAT for Storybook Bot and update the `PAT_STORYBOOK_BOT
` secret of this repository
 - Make the repo public
 - Add links in the doc? website?
 
## How to test

- A GitHub action called `Generate And Push Repros` should have run without any error: https://github.com/storybookjs/storybook/pull/15877/checks?check_run_id=3393520779
- Repository with all the repros should have been updated: https://github.com/storybookjs/repro-templates

### CODE REVIEW

File: .github/workflows/generate-repros.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Setup git user
+        run: |
+          git config --global user.name "$(git --no-pager log --format=format:'%an' -n 1)"
+          git config --global user.email "$(git --no-pager log --format=format:'%ae' -n 1)"
+      - name: Install dependencies
+        run: yarn install
+      - name: Generate repros with Latest Storybook CLI
+        run: yarn generate-repros --remote=https://gaetanmaisse:${{ secrets.PAT_STORYBOOK_BOT}}@github.com/storybookjs/repro-templates.git --push --force-push
+      - name: Generate repros with Next Storybook CLI
+        run: yarn generate-repros --next --remote=https://gaetanmaisse:${{ secrets.PAT_STORYBOOK_BOT}}@github.com/storybookjs/repro-templates.git --push --force-push
```
Comment (gaetanmaisse) [inline .github/workflows/generate-repros.yml:28]: Update user

File: scripts/repros-generator/templates/next/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+![Storybook Latest Badge](https://img.shields.io/npm/v/@storybook/react/next)
+
+The repros have been generated with the latest stable version of Storybook
+
+Preview any repro live on [StackBlitz](http://stackblitz.com/):
+
+- [Angular](https://stackblitz.com/github/storybookjs/repro-templates/tree/next/angular?preset=node)
+- [React](https://stackblitz.com/github/storybookjs/repro-templates/tree/next/react?preset=node)
+- [Create React App](https://stackblitz.com/github/storybookjs/repro-templates/tree/next/cra?preset=node)
+- [Create React App + TypeScript](https://stackblitz.com/github/{{}repo}/tree/next/cra_typescript?preset=node)
```
Comment (shilman) [inline scripts/repros-generator/templates/next/README.md:12]: @gaetanmaisse  what's going on with `{{}repo}`?

Comment (gaetanmaisse) [reply scripts/repros-generator/templates/next/README.md]: Fixed ;)

File: .github/workflows/generate-repros.yml (hunk 1 above)
Comment (gaetanmaisse) [reply .github/workflows/generate-repros.yml]: Done

**Review by shilman — APPROVED** [review]
LGTM! great stuff as usual @gaetanmaisse 🚀

**Pull request merged**